spec/prog/populate_ipv4_cache_spec.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require_relative "../model/spec_helper"
+
+RSpec.describe Prog::PopulateIpv4Cache do
+  subject(:pc) {
+    described_class.new(Strand.new(prog: "PopulateIpv4Cache"))
+  }
+
+  describe "#wait" do
+    it "populates file and naps an hour" do
```

Comment:
This approach doesn't work.  The prog runs in a single respirate process, and only that process will have the file written.  None of the web servers would have that file created (the file systems are not shared between dynos in production).

You should switch to storing the cached information in the database.